### ISSUE TITLE
custom op  error

### ISSUE THREAD

**Issue by ZHUANG-JLU** [tensorflow/tensorflow#57742]
To replace the tf.py_func, I wrote custom op by c++ and use tf.load_op_library to load the op in python.And get Segmentation fault (core dumped).And when  I comment  SetShapeFn, there is no error. But the output shape of the custom op is unknow.Could someone can help me ?  Another info is using pb model to inference by c++ custom op is correct. Below is the error and code
######################################################################################################
This is the error.
######################################################################################################
WARNING:tensorflow:From` /home/zhj/Py/frustum-pointnets-master/models/tf_util.py:384: The name tf.nn.max_pool is deprecated. Please use tf.nn.max_pool2d instead.

WARNING:tensorflow:From /home/zhj/Py/frustum-pointnets-master/models/tf_util.py:613: calling dropout (from tensorflow.python.ops.nn_ops) with keep_prob is deprecated and will be removed in a future version.
Instructions for updating:
Please use `rate` instead of `keep_prob`. Rate should be set to `rate = 1 - keep_prob`.
WARNING:tensorflow:From /home/zhj/Py/frustum-pointnets-master/models/model_util.py:235: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.cast` instead.
WARNING:tensorflow:From /home/zhj/Py/frustum-pointnets-master/models/model_util.py:236: calling reduce_sum_v1 (from tensorflow.python.ops.math_ops) with keep_dims is deprecated and will be removed in a future version.
Instructions for updating:
keep_dims is deprecated, use keepdims instead
mask.shape---------------------------- (1, 1024)
Segmentation fault (core dumped)

######################################################################################################
This is the custom op 

```
#include "tensorflow/core/framework/common_shape_fns.h"
#include "tensorflow/core/framework/op.h"
#include "tensorflow/core/framework/op_kernel.h"
#include "tensorflow/core/framework/shape_inference.h"

namespace tensorflow {

using shape_inference::DimensionHandle;
using shape_inference::InferenceContext;
using shape_inference::ShapeHandle;
using shape_inference::UnchangedShape;

REGISTER_OP("Zhj")
    .Input("input: float")
    .Output("output: int32")
    .SetShapeFn([](::tensorflow::shape_inference::InferenceContext* c) {
        DimensionHandle batch = c->Dim(c->input(0), 0);
        ShapeHandle output_shape = c->MakeShape({batch,c->MakeDim(512),c->MakeDim(2)});
        c->set_output(0, output_shape);
        return Status::OK();
    });

class PyFuncZhj : public tensorflow::OpKernel {
 public:
  explicit PyFuncZhj(OpKernelConstruction* context) : OpKernel(context) {}

  void Compute(OpKernelContext* context) override {
    // Grab the input tensor
    const Tensor& tensor_in = context->input(0);
    const TensorShape& input_shape = tensor_in.shape();
    auto inputEigen = tensor_in.shaped<float, 2>({input_shape.dim_size(0), input_shape.dim_size(1)});
    // Create an output tensor
    Tensor* output_tensor;
    OP_REQUIRES_OK(context, context->allocate_output(
                                0, TensorShape({input_shape.dim_size(0), 512,2}),
                                &output_tensor));
    auto outputEigen = output_tensor->shaped<int32, 3>({input_shape.dim_size(0), 512, 2});
    std::vector<std::vector<int>> indicesVec;
    for (int i = 0; i < input_shape.dim_size(0); i++)
    {
      std::vector<int> tmpIndices;
      for (int j = 0; j < input_shape.dim_size(1); j++)//output_shape.dim_size(1)
      {
        if(inputEigen(i,j) > 0.1)
        {
          tmpIndices.push_back(j);
        }
      }
      int lengthIndices = tmpIndices.size();
      if(lengthIndices > 0)
      {
        while(tmpIndices.size() < 512)
        {
          int rand_num = rand() % lengthIndices;
          tmpIndices.push_back(tmpIndices[rand_num]);
        }
        while (tmpIndices.size() > 512)
        {
          int rand_num = rand() % tmpIndices.size();
          tmpIndices.erase(tmpIndices.begin() + rand_num);
        }
      }
      indicesVec.push_back(tmpIndices);
    }
    for(int i = 0; i < indicesVec.size(); i++)
    {
      if(0 == indicesVec[i].size())
      {
        for(int j = 0; j < 512; j++)
        {
          outputEigen(i, j, 0) = i;  // batchsize
          outputEigen(i, j, 1) = 0;
        }
      }
      else
      {
        for(int j = 0; j < indicesVec[i].size(); j++)
        {
          outputEigen(i, j, 0) = i;  // batchsize
          outputEigen(i, j, 1) = indicesVec[i][j];
        }
      }
    }
  }
};
REGISTER_KERNEL_BUILDER(Name("Zhj").Device(DEVICE_CPU), PyFuncZhj);
}
```

######################################################################################################
This is the python code
```
Zhj = tf.load_op_library("/home/zhj/Py/frustum-pointnets-master/train/log_v1/zhj.so")
def tf_gather_object_pc(point_cloud, mask, npoints=512):
    ''' Gather object point clouds according to predicted masks.
    Input:
        point_cloud: TF tensor in shape (B,N,C)
        mask: TF tensor in shape (B,N) of 0 (not pick) or 1 (pick)
        npoints: int scalar, maximum number of points to keep (default: 512)
    Output:
        object_pc: TF tensor in shape (B,npoint,C)
        indices: TF int tensor in shape (B,npoint,2)
    '''
    def mask_to_indices(mask):
        indices = np.zeros((mask.shape[0], npoints, 2), dtype=np.int32)
        for i in range(mask.shape[0]):
            pos_indices = np.where(mask[i,:]>0.5)[0]
            # skip cases when pos_indices is empty
            if len(pos_indices) > 0: 
                if len(pos_indices) > npoints:
                    choice = np.random.choice(len(pos_indices),
                        npoints, replace=False)
                else:
                    choice = np.random.choice(len(pos_indices),
                        npoints-len(pos_indices), replace=True)
                    choice = np.concatenate((np.arange(len(pos_indices)), choice))
                np.random.shuffle(choice)
                indices[i,:,1] = pos_indices[choice]
            indices[i,:,0] = i
        return indices

    print("mask.shape----------------------------", mask.shape)  # mask.shape: (1,1024)
    # indices = tf.py_func(mask_to_indices, [mask], tf.int32)
    indices = Zhj.zhj([mask])
    object_pc = tf.gather_nd(point_cloud, indices)
    return object_pc, indices
```

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
